pkg/services/grafana-apiserver/service.go

```diff
@@ -158,11 +158,14 @@ func ProvideService(
 				req.URL.Path = "/metrics"
 			}
 
+			// TODO: none of these are used today, we use the context directly in k8s
 			ctx := req.Context()
 			signedInUser := appcontext.MustUser(ctx)
-
 			req.Header.Set("X-Remote-User", strconv.FormatInt(signedInUser.UserID, 10))
 			req.Header.Set("X-Remote-Group", "grafana")
+			if signedInUser.IDToken != "" {
+				req.Header.Set("X-Remote-Extra-ID-Token", signedInUser.IDToken)
```

Comment:
this is only used in the context where we already have a user -- IIUC, we need to add the special logic inside:
https://github.com/grafana/grafana/blob/main/pkg/services/grafana-apiserver/service.go#L290

that converts the k8s user to grafana signed in user

Reply:
@ryantxu i just talked this over with @DanCech and i'll push that update to this PR